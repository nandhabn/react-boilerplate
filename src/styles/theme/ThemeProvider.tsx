import React from 'react';
import { useSelector } from 'react-redux';
import { useThemeSlice } from './slice';
import { selectTheme } from './slice/selectors';

export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice();

  // FIXME -  add theme to the childrens
  const theme = useSelector(selectTheme);
  return <div>{React.Children.only(props.children)}</div>;
};
