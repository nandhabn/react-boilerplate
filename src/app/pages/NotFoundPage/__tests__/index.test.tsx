import * as React from 'react';
import { NotFoundPage } from '..';
import { MemoryRouter } from 'react-router-dom';
import { themes } from 'styles/theme/themes';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';

const renderPage = () =>
  renderer.create(
    <MemoryRouter>
      <HelmetProvider>
        <NotFoundPage />
      </HelmetProvider>
    </MemoryRouter>,
  );

describe('<NotFoundPage />', () => {
  it('should match snapshot', () => {
    const notFoundPage = renderPage();
    expect(notFoundPage.toJSON()).toMatchSnapshot();
  });

  it('should should contain Link', () => {
    const notFoundPage = renderPage();
  });
});
