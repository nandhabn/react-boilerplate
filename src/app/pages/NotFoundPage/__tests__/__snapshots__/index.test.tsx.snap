// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match snapshot 1`] = `
Array [
  "4",
  <span
    aria-label="Crying Face"
    role="img"
  >
    ðŸ˜¢
  </span>,
  "4",
  <p>
    Page not found.
  </p>,
]
`;
